src/lib/import_export/binary/binary_parser.cpp

```diff
@@ -300,17 +304,9 @@ std::unique_ptr<const BaseCompressedVector> BinaryParser::_import_offset_value_v
 
 std::shared_ptr<FixedStringVector> BinaryParser::_import_fixed_string_vector(std::ifstream& file, const size_t count) {
   const auto string_length = _read_value<uint32_t>(file);
-  std::vector<pmr_string> values(count);
-
-  for (uint32_t row = 0; row < count; row++) {
-    std::ostringstream value;
-    for (uint32_t c = 0; c < string_length; c++) {
-      value << _read_value<char>(file);
-    }
-    values[row] = pmr_string(value.str());
-  }
-
-  return std::make_shared<FixedStringVector>(values.begin(), values.end(), string_length);
+  pmr_vector<char> values(string_length * count);
+  file.read(values.data(), values.size());
+  return std::make_shared<FixedStringVector>(values, string_length);
```

Comment:
`std::move(values)`

Reply:
I had a look at the FixedSizeByteAlignedVector constructor and, like there, did the move inside the value assignment in the constructor.

Reply:
Not sure what you are saying. The constructor takes a `pmr_vector<char> chars`, which is a call by value. If you don't move `values` in, it will be copied.

Whatever happens inside the constructor is a completely different thing.

Reply:
Ah, okay, I added this. So the compressed vectors of the other segments are copied while reading binary then?

Reply:
Could you give me a line number, please? If you are talking about this:

https://github.com/hyrise/hyrise/blob/4b842df725c16fe24c536da8ea0ef8994d7e568d/src/lib/import_export/binary/binary_parser.cpp#L295

nothing is copied because _read_values is a temporary (see: guaranteed copy elision).

If you are talking about this:

https://github.com/hyrise/hyrise/blob/4b842df725c16fe24c536da8ea0ef8994d7e568d/src/lib/import_export/binary/binary_parser.cpp#L200-L203

`null_values` is copied, but as it is a shared_ptr, it is by far not as bad as copying the actual vector.

Reply:
Okay, I see :)